hal/sparcv8leon/gaisler/irqamp.c

```diff
@@ -82,23 +90,37 @@ extern int threads_schedule(unsigned int n, cpu_context_t *context, void *arg);
 
 void hal_cpuBroadcastIPI(unsigned int intr)
 {
+	unsigned int id = hal_cpuGetID(), i;
+
+	for (i = 0; i < hal_cpuGetCount(); ++i) {
```

Comment:
Nitpick - would be better to cache this value, code as is will call the function on every iteration